imap_processing/spice/geometry.py

```diff
@@ -235,6 +235,74 @@ def get_spacecraft_spin_phase(
     return spin_phases
 
 
+def get_instrument_spin_phase(
+    query_met_times: Union[float, npt.NDArray],
+    instrument: SpiceFrame,
+) -> Union[float, npt.NDArray]:
+    """
+    Get the instrument spin phase for the input query times.
+
+    Formula to calculate spin phase:
+        instrument_spin_phase = (spacecraft_spin_phase + instrument_spin_offset) % 1
+
+    Parameters
+    ----------
+    query_met_times : float or np.ndarray
+        Query times in Mission Elapsed Time (MET).
+    instrument : SpiceFrame
+        Instrument frame to calculate spin phase for.
+
+    Returns
+    -------
+    spin_phase : float or np.ndarray
+        Instrument spin phase for the input query times. Spin phase is a
+        floating point number in the range [0, 1) corresponding to the
+        spin angle / 360.
+    """
+    spacecraft_spin_phase = get_spacecraft_spin_phase(query_met_times)
+    instrument_spin_phase_offset = get_spacecraft_to_instrument_spin_phase_offset(
+        instrument
+    )
+    return (spacecraft_spin_phase + instrument_spin_phase_offset) % 1
```

Comment:
This `% 1` is to get the `[0, 1)` range?

Reply:
Yes, that is correct. The s/c spin phase is in `[0, 1)` but then we add a positive offset value that is also in the range [0, 1), so we need to "unwind" the phase back into the `[0, 1)` range.